app/services/jwt_token.rb

```diff
@@ -34,8 +34,21 @@ def to_s
   end
 
   # This method does not verify if the token signature is valid
+  # defined? keyword checks if expression is currently defined. It
+  # therefore, helps to memoize the nil JWT values.
   def decoded_payload
-    @decoded_payload ||= JWT.decode(token, nil, false).first
+    return @decoded_payload if defined?(@decoded_payload)
+    @decoded_payload = JWT.decode(token, nil, false).first
+
+    unless @decoded_payload.is_a?(Hash)
+      logger.debug "Received payload: #{@decoded_payload}"
+      @decoded_payload = nil
+      logger.error "Invalid decoded JWT format."
```

Comment:
not critical, but i think this order might be more useful:
```suggestion
      logger.error "Invalid decoded JWT format."
      logger.debug "Received payload: #{@decoded_payload}"
      @decoded_payload = nil
```